### PR TITLE
Schema endpoint validation

### PR DESCRIPTION
**Pull request by walkingtowork**
@rmutter @baylee 
https://github.com/yeti/signals/issues/38

- Added path validation to the `--datamodels` flag in `main.py`
- Added `validate_apis_and_objects` function to `schema.py`
  - Loop over the endpoints for each `url` in the schema, then search the `objects` dict for matches of `parameters`, `requests`, and `responses`, and throw an `error` if a match isn't found.

### CODE REVIEW

File: signals/parser/schema.py [hunk 1]
```diff
@@ -28,6 +29,25 @@ def create_objects(self, object_json):
             data_object = DataObject(object_name, data_object_json[object_name])
             self.data_objects[data_object.name] = data_object
 
+    # Note: Why do we call them "apis" in method names, but then call them "urls" in the schema/code
```
Comment (rmutter) [inline signals/parser/schema.py]: A single API is called an endpoint here - a URL is a collection of endpoints: ex. we have /api/posts/ which could have a GetAPI, PostAPI, and DeleteAPI

Agree we could have some better consistent naming

File: signals/parser/schema.py [hunk 2]
```diff
@@ -28,6 +29,25 @@ def create_objects(self, object_json):
...
+    def validate_apis_and_objects(self):
+        for url in self.urls:
+            for endpoint in URL.URL_ENDPOINTS.keys():
+                api = getattr(url, endpoint, None)
+                if api:
+                    if getattr(api, 'parameters_object', None):
```
Comment (rmutter) [inline signals/parser/schema.py]: We should reduce all of these `if`'s to a single call that calls another method and passes in the object attribute we're looking for. Even better we have a list of them `['parameters_object', 'request_object', 'response_object']` and we loop through calling that method on each of them.

File: signals/parser/api.py [hunk 3]
```diff
@@ -33,6 +33,15 @@ def process_authorization_attribute(self, attribute):
             error_message = "Found invalid authorization attribute: {} for {}, exiting."
             raise SignalsError(error_message.format(attribute, self.url_path))
 
+    # def validate(self, endpoint_json):
```
Comment (rmutter) [inline signals/parser/api.py]: I'm sure you know, but lots of comments to clean up in this file.

File: signals/parser/schema.py (hunk 1 above)
Comment (rmutter) [inline signals/parser/schema.py]: Also, I'm assuming you did this out here because you didn't have access to the `data_objects` inside of the API objects?

**Comment by rmutter** [comment]
@walkingtowork some comments + tests failed

File: signals/parser/schema.py (hunk 1 above)
Comment (walkingtowork) [inline signals/parser/schema.py]: Yup.

File: signals/parser/schema.py (hunk 2 above)
Comment (walkingtowork) [inline signals/parser/schema.py]: I wanted to do that, but if I write `if api['parameters_object'] not in self.data_objects:` I get an error.

File: signals/parser/api.py (hunk 3 above)
Comment (walkingtowork) [inline signals/parser/api.py]: Yeah, just didn't want to delete those until I got the OK to leave my code in the schema.py file.

**Pull request merged**